docs/src/architecture/accounts.md

```diff
@@ -8,44 +8,51 @@ An account is an entity which holds assets and defines rules of how these assets
 
-### Regular Account with updatable code
+* Alice grinds a new Account ID according to the account types (see below) using a wallet
+* Alice's Miden Client requests the Miden Node if new Account ID already exists
+* Alice forwards the new Account ID to Bob, e.g. when Alice wants to receive funds
+* Bob executes a transaction and creates a note that contains an asset for Alice
+* Alice consumes Bob's note to receive the asset in a transaction
+  * if local transaction: She executes and proves the transaction, then she sends proof and Account ID plus potential data to Operator
+  * if network transaction: She prepares the transaction and sends transaction data incl. Account ID plus potential data to Operator 
+* Operator receives new Account ID and - if transaction is correct - adds the ID to the Account DB
+
+
+## Types of accounts
+There are four types of accounts in Miden. The first two bits of the Account ID specify the type of the account.
+
+### Regular account with updatable code
 This account type will be used by most users. They can specify and change their code and use this account for a wallet. The Account ID will start with `00`.
 
-### Regular Account with immutable code
+### Regular account with immutable code
 This account type will be used by most regular smart contracts. Once deployed the code should not change and no one should be able to change it. The Account ID will start with `01`.
 
 ### Fungible asset faucet with immutable code
-Assets need to be created by Accounts in Miden. The Account ID will start with `10`. 
+Assets need to be created by accounts in Miden. The Account ID will start with `10`. 
 
 ### Non-fungible asset faucet with immutable code
-Assets need to be created by Accounts in Miden. The Account ID will start with `11`. 
+Assets need to be created by accounts in Miden. The Account ID will start with `11`. 
 
 
 ## Account data storage
-Account data can be public or private.
+Account data - stored by the Miden Node - can be public or private.
 
 * Accounts with public state, where the actual state is stored on chain. These would be similar to how accounts work in public blockchains.
-* Accounts with private state, where only the hash of the account is stored on chain. The hash is defined as `hash([account ID], [storage root], [vault root], [code root])`.
-
-The third most significant bit of the ID specifies whether the account data is public (0) or private (1).
-
-Missing: 
+* Accounts with private state, where only the hash of the account is stored on chain. The hash is defined as: \
+`hash([account ID, 0, 0, nonce], [vault root], [storage root], [code root])`
 
-* How are accounts being created?
-* Life cycle of an account?
-* Account Wallet interface ?
-* Account IDs
+The third most significant bit of the ID specifies whether the account data is public `0` or private `1`.
```

Comment:
It may be good to add a diagram of account ID somewhere in this section (could be something super simple illustrating how the first 3 bits have semantic meaning). We could also show that if u64 is used to represent an account, the last 4 bits would be ignored (if #104 is implemented).